includes/User.php

```diff
@@ -4837,9 +4837,28 @@ public static function makeGroupLinkWiki( $group, $text = '' ) {
 	 * @param $right String Right to query
 	 * @return String Localized description of the right
 	 */
-	public static function getRightDescription( $right ) {
+    public static function getRightDescription( $right ) {
 		$key = "right-$right";
 		$msg = wfMessage( $key );
 		return $msg->isBlank() ? $right : $msg->text();
 	}
+
+
+	/**
+	 * We want to use one source for username.
+	 * This function will perform the lookup if
+     * $wgEnableUsernameLookup is true
+	 *
+	 * @param $userId int userId
+	 * @param $name string anon username
+	 * @return string
+	 */
+	public static function getUsername( $userId, $name ) {
+		global $wgEnableUsernameLookup;
+		// we only want to perform the lookup if it's enabled and the user is not an anon
+		if( $wgEnableUsernameLookup && $userId != 0 ) {
+            return static::whoIs($userId);
```

Comment:
If user with given id is not found function will return false. Is that the desired behaviour? Or should we then return the $name parameter. We were discussing that the $name parameter should be a "default" value, so that's not how it works now. Not saying it's wrong, just want to make sure what's expected. I would also document that behaviour in the comment to the function.